primary.c

```diff
@@ -14,17 +17,49 @@ bool isPrime(int number){
 }
 
 int main(int argc, char *argv[]){
-	if(argc == 1){
-		printf("You don't type any number, Please type numberes throw argv");
-		return 0;
-	}
-	int num;
-	int i = 0;
-	for(i = 1; i < argc; i++){
-		num = atoi(argv[i]);
-		if(isPrime(num)){
-			printf("%d ", num);
+	        int fd[2];
+        pid_t p;
+
+        if (pipe(fd) == -1){
+                fprintf(stderr, "Pipe failed");
+                return 1;
+        }
+
+        p = fork();
+
+        if (p < 0){
+                fprintf(stderr, "fork failed");
+                return 1;
+        }
+
+        if (p == 0){
+                close(fd[0]);
+
+                if(argc == 1){
+                    printf("You don't type any number, Please type numberes throw argv");
+                    return 0;
+                }
+                int num;
+                int i = 0;
+                for(i = 1; i < argc; i++){
+                       write(fd[1], argv[i], strlen(argv[i]));
+                }
+                close(fd[1]);
+        }
+
+        else{
```

Comment:
The child process should `execve` another executable that checks for prime numbers.
That means you should move the `IsPrime` function to another file.
You can pass the numbers through the argv by just calling `execve("./real_is_prime", argv, NULL)`
I want you to use `dup2(1, pipe_fd)` so that `./real_is_prime` can write to `stdout` and the parent process reads the numbers.